drivers/lora/shell.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2020 Andreas Sandberg
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/lora.h>
+#include <inttypes.h>
+#include <shell/shell.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+
+LOG_MODULE_REGISTER(lora_shell);
+
+#define DEFAULT_RADIO_NODE DT_ALIAS(lora0)
+BUILD_ASSERT(DT_HAS_NODE(DEFAULT_RADIO_NODE),
+	     "No default LoRa radio specified in DT");
+#define DEFAULT_RADIO DT_LABEL(DEFAULT_RADIO_NODE)
+
+static struct lora_modem_config modem_config = {
+	.frequency = 0,
+	.bandwidth = BW_125_KHZ,
+	.datarate = SF_10,
+	.coding_rate = CR_4_5,
+	.preamble_len = 8,
+	.tx_power = 4,
+};
+
+static const int bw_table[] = {
+	[BW_125_KHZ] = 125,
+	[BW_250_KHZ] = 250,
+	[BW_500_KHZ] = 500,
+};
+
+static struct device *get_modem(const struct shell *shell,
+				struct lora_modem_config *cfg)
+{
+	struct device *dev;
+	int ret;
+
+	dev = device_get_binding(DEFAULT_RADIO);
+	if (!dev) {
+		shell_error(shell, "%s Device not found", DEFAULT_RADIO);
+		return NULL;
+	}
+
+	if (cfg) {
```

Comment:
I don't like the fact that `get_modem` applies config. I would rather call `device_get_binding()` separately and create a helper function for applying lora config (and checking inside `cfg->frequency == 0` a you do here).